Test/Unit/Helper/IpAddressTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2020 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Tests\Helper;
+
+class IpAddressTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var \Adyen\Payment\Helper\IpAddress
+     */
+    private $ipAddressHelper;
+
+    private function getSimpleMock($originalClassName)
+    {
+        return $this->getMockBuilder($originalClassName)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    protected function setUp()
+    {
+        $cache = $this->getSimpleMock(\Magento\Framework\App\CacheInterface::class);
+        $serializer = $this->getSimpleMock(\Magento\Framework\Serialize\SerializerInterface::class);
+        $ipAddressUtil = $this->getSimpleMock(\Adyen\Util\IpAddress::class);
+
+        $this->ipAddressHelper = new \Adyen\Payment\Helper\IpAddress(
+            $ipAddressUtil,
+            $cache,
+            $serializer
+        );
+    }
+
+    public function testIsIpAddressValid()
+    {
+        $this->assertTrue(is_bool($this->ipAddressHelper->isIpAddressValid([gethostbyname('outgoing1.adyen.com')])));
```

Comment:
This test is not unitary and will depend on the network layer. I'd just feed the `isIpAddressValid` method with a mocked array of results (like a couple of IP addresses)

Also, we could cover more cases, such as:
* IP addresses that are not setup within Adyen
* invalid input for IP address

Reply:
Thank you @msilvagarcia. I made some adjustments to the test. I'm mocking the result of some external functions and provide a set of IP addresses to check, including an invalid one. What do you think?